File: frontend/src/main/java/com/example/callcenter/Entity/ContactStatus.java
package com.example.callcenter.Entity;

import lombok.Getter;

@Getter
public enum ContactStatus {
    NOT_CONTACTED("Not Contacted"),
    CONTACTED_AVAILABLE("Contacted - Available"),
    CONTACTED_UNAVAILABLE("Contacted - Unavailable"),
    NO_ANSWER("No Answer"),
    WRONG_NUMBER("Wrong Number"),
    CALL_BACK_LATER("Call Back Later");

    private final String label;

    ContactStatus(String label) {
        this.label = label;
    }
} 

File: frontend/src/app/models/ContactStatus.ts
export enum ContactStatus {
    NOT_CONTACTED = 'NOT_CONTACTED',
    CONTACTED_AVAILABLE = 'CONTACTED_AVAILABLE',
    CONTACTED_UNAVAILABLE = 'CONTACTED_UNAVAILABLE',
    NO_ANSWER = 'NO_ANSWER',
    WRONG_NUMBER = 'WRONG_NUMBER',
    CALL_BACK_LATER = 'CALL_BACK_LATER'
}

export const getContactStatusLabel = (status: ContactStatus): string => {
    return status.replace(/_/g, ' ')
        .toLowerCase()
        .split(' ')
        .map(word => word.charAt(0).toUpperCase() + word.slice(1))
        .join(' ');
}; 